test/extended/util/framework.go

```diff
@@ -1691,15 +1691,16 @@ type podExecutor struct {
 
 // NewPodExecutor returns an executor capable of running commands in a Pod.
 func NewPodExecutor(oc *CLI, name, image string) (*podExecutor, error) {
-	out, err := oc.Run("run").Args(name, "--labels", "name="+name, "--image", image, "--restart", "Never", "--command", "--", "/bin/bash", "-c", "sleep infinity").Output()
```

Comment:
Is it your intention to revert to the default restart policy (which is "Always") instead of using "Never"?

Reply:
Good point, not directly my intention, but I'd argue it's a wash between the two, however, `Always` may be slightly better.

`Always` will always check if the image changed in the remote repository, and replace it if it did. If the remote image didn't change, then it does not re-download. So we shouldn't be pulling anymore than previously with `Never`. Thoughts?

Reply:
Are you confusing `ImagePullPolicy` with `RestartPolicy`?  I expect we want the default `ImagePullPolicy` (which is "Always" when the image tag is "latest" and "IfNotPresent" otherwise) and `RestartPolicy: Never` (i.e. never restart the pod).

Reply:
Whoops yes, sorry about that.

I tested by putting a syntax error in the pod command with the default `ImagePullPolicy` of `Always` and it successfully passed `CreateExecPodOrFail`, since I suppose the polling function `WaitForPodNameRunningInNamespace` caught it in a `Running` state as it was about to fail. Either way, it failed further down the code when trying to `Exec` on it.

I switched to `Never` and `CreateExecPodOrFail` didn't pass with the syntax error. Seems better for it to fail faster. I feel `Always` also convoluted the debugging process with things constantly getting restarted.

Pushing update.